controllers/configurationpolicy_controller.go

```diff
@@ -1598,6 +1601,15 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 			} else {
 				resultEvent.compliant = false
 				resultEvent.reason = reasonWantFoundDNE
+				// Length of objNames = 0, complianceType == musthave and the name field is empty
+				// Find Noncompliant resources to add to the status.relatedObjects
+				if objDetails.kind != "" && objDetails.name == "" {
+					// Change reason to Resource found but does not match
+					if len(allResourceNames) > 0 {
+						resultEvent.reason = reasonWantFoundNoMatch
+						shouldAddCondensedRelatedObj = true
```

Comment:
I think you want to always set this to true if the object template was unnamed except the special case of only one object returned from `getNamesOfKind` which is handled by `handleSingleObj` above.

I think this current code won't add the correct related object entry when no objects were returned from `getNamesOfKind`.

Reply:
reason is `Resource found but does not match` To me, it doesn't make sense if there is 0 found resource

Reply:
I was specifically referring to `shouldAddCondensedRelatedObj`.

Reply:
When `allResourceNames = 0`, don't need to add related obj `Resource found but does not match` doesn't match at all

Reply:
Right... but in that case you won't have a related objects entry. I'm saying that when `allResourceNames != 1` (1 object is handled by `handleSingleObj` above), then `shouldAddCondensedRelatedObj = true` should be set to true when doing `musthave` and it's not compliant. The reason is different based on if `allResourceNames > 1`

Reply:
I thought `resultEvent.compliant = false` mean objNames is 0, allResourceNames doesn't include matched obj